src/uct/cuda/cuda_ipc/cuda_ipc_cache.c

```diff
@@ -280,21 +286,19 @@ uct_cuda_ipc_open_memhandle_mempool(const uct_cuda_ipc_rkey_t *key,
 static ucs_status_t uct_cuda_ipc_open_memhandle(const uct_cuda_ipc_rkey_t *key,
                                                 CUdeviceptr *mapped_addr)
 {
+    CUipcMemHandle *legacy_handle = (CUipcMemHandle*)&key->ph;
+
 #if HAVE_CUDA_FABRIC
     ucs_trace("key handle type %u", key->ph.handle_type);
+    legacy_handle = (CUipcMemHandle*)&key->ph.handle.legacy;
 
-    if (key->ph.handle_type == UCT_CUDA_IPC_KEY_HANDLE_TYPE_LEGACY) {
-        return uct_cuda_ipc_open_memhandle_legacy(key->ph.handle.legacy, mapped_addr);
-    } else if (key->ph.handle_type == UCT_CUDA_IPC_KEY_HANDLE_TYPE_VMM) {
+    if (key->ph.handle_type == UCT_CUDA_IPC_KEY_HANDLE_TYPE_VMM) {
         return uct_cuda_ipc_open_memhandle_vmm(key, mapped_addr);
     } else if (key->ph.handle_type == UCT_CUDA_IPC_KEY_HANDLE_TYPE_MEMPOOL) {
         return uct_cuda_ipc_open_memhandle_mempool(key, mapped_addr);
-    } else {
-        return UCS_ERR_NO_RESOURCE;
-    }
-#else
-    return uct_cuda_ipc_open_memhandle_legacy(key->ph, mapped_addr);
+    } else
 #endif
+    return uct_cuda_ipc_open_memhandle_legacy(*legacy_handle, mapped_addr);
```

Comment:
actually better revert, I didn't notice legacy_handle location is different depending on HAVE_CUDA_FABRIC